crud_for_code_review.py

```diff
@@ -0,0 +1,546 @@
+################################################################################
+
+def create_track(track_id, track_name, artist_name):
+    """Create and return a new Track"""
+
+    track = Track(track_id=track_id,
+                track_name=track_name,
+                artist_name=artist_name)
+
+    db.session.add(track)
+    db.session.commit()
+
+    return track
+
+
+def create_user_track(user_id, track_id):
+    """Create and return a new UserTrack"""
+
+    user_track = UserTrack(user_id=user_id, track_id=track_id)
+
+    db.session.add(user_track)
+    db.session.commit()
+
+    return user_track
+
+
+def get_track_by_id(track_id):
+    """Query tracks table and return object"""
+
+    return Track.query.get(track_id)
+
+
+def check_audio(track_id):
+    """Query audio table and return object"""
+
+    return Audio.query.filter_by(track_id=track_id).first()
+
+
+def get_user_tracks_list(user_id):
+    """Get user tracks list to pass into API call for audio features"""
+
+    user = User.query.get(user_id)
+
+    track_list = []
+    for track in user.tracks:
+        track_list.append(track.track_id)
+
+    return track_list
+
+
+def check_user_tracks(user_id, track_id):
+    """Check if a track is in user_tracks"""
+
+    return db.session.query(UserTrack).filter((UserTrack.user_id==user_id), (UserTrack.track_id==track_id)).first()
+
+
+def avg_audio_features(user_id):
+    """Returns list of average audio features for user's top 50 tracks"""
+
+    # user_join = User.query.options( 
+    #      db.joinedload('tracks')
+    #        .joinedload('audio')  
+    #     ).get(user_id)
+
+    audio = []
+
+    # #data being used in radar chart, pass variables in order of expected labels
+    avg_dance = db.session.query(db.func.avg(Audio.danceability)).filter((UserTrack.user_id==user_id), (UserTrack.track_id == Audio.track_id)).scalar()
+    avg_energy = db.session.query(db.func.avg(Audio.energy)).filter((UserTrack.user_id==user_id), (UserTrack.track_id == Audio.track_id)).scalar()
+    avg_speech = db.session.query(db.func.avg(Audio.speechiness)).filter((UserTrack.user_id==user_id), (UserTrack.track_id == Audio.track_id)).scalar()
+    avg_acoustic = db.session.query(db.func.avg(Audio.acousticness)).filter((UserTrack.user_id==user_id), (UserTrack.track_id == Audio.track_id)).scalar()
+    avg_instrumental = db.session.query(db.func.avg(Audio.instrumentalness)).filter((UserTrack.user_id==user_id), (UserTrack.track_id == Audio.track_id)).scalar()
+    avg_liveness = db.session.query(db.func.avg(Audio.liveness)).filter((UserTrack.user_id==user_id), (UserTrack.track_id == Audio.track_id)).scalar()
+    avg_valence = db.session.query(db.func.avg(Audio.valence)).filter((UserTrack.user_id==user_id), (UserTrack.track_id == Audio.track_id)).scalar()
+
+    audio.extend((avg_dance, avg_energy, avg_speech, avg_acoustic, avg_instrumental, avg_liveness, avg_valence))
+    
+    return audio
+
+
+def audio_stats(user_id):
```

Comment:
This function and some others above aren't doing crud, they're more manipulating data, so they could be in their own module, especially since crud.py is getting large.